unit/util/simplify_expr.cpp

```diff
@@ -61,7 +61,10 @@ TEST_CASE("Simplify const pointer offset")
   REQUIRE(offset_value==1234);
 }
 
-TEST_CASE("Simplify Java boolean -> int -> boolean casts")
+namespace
+{
+
+void test_unnecessary_cast(std::function<typet()> get_type_func)
```

Comment:
Why not just take a `const typet &` as the input here?

Reply:
Good point